src/sources/statsd/mod.rs

```diff
@@ -130,12 +212,88 @@ mod test {
     }
 
     #[tokio::test]
-    async fn test_statsd() {
+    async fn test_statsd_udp() {
         let in_addr = next_addr();
+        let config = StatsdConfig::Udp(UdpConfig { address: in_addr });
+        let sender = {
+            let (sender, mut receiver) = mpsc::channel(200);
+            let addr = in_addr;
+            tokio::spawn(async move {
+                let bind_addr = next_addr();
+                let mut socket = tokio::net::UdpSocket::bind(bind_addr).await.unwrap();
+                socket.connect(addr).await.unwrap();
+                while let Some(bytes) = receiver.recv().await {
+                    socket.send(bytes).await.unwrap();
+                }
+            });
+            sender
+        };
+        test_statsd(config, sender).await;
+    }
+
+    #[tokio::test]
+    async fn test_statsd_tcp() {
+        let in_addr = next_addr();
+        let config = StatsdConfig::Tcp(TcpConfig {
+            address: in_addr.into(),
+            tls: None,
+            shutdown_timeout_secs: 30,
+        });
+        let sender = {
+            let (sender, mut receiver) = mpsc::channel(200);
+            let addr = in_addr;
+            tokio::spawn(async move {
+                while let Some(bytes) = receiver.recv().await {
+                    use tokio::io::AsyncWriteExt;
```

Comment:
Can you move this import to the global `mod test`? (same for `test_statsd_unix`)